src/main/c/Jep/convert_p2j.c

```diff
@@ -865,6 +865,16 @@ jobject PyObject_As_jobject(JNIEnv *env, PyObject *pyobject,
         if ((*env)->IsAssignableFrom(env, pyjobject->clazz, expectedType)) {
             return (*env)->NewLocalRef(env, pyjobject->object);
         }
+    } else if (PyCallable_Check(pyobject)) {
+        if (isFunctionalInterfaceType(env, expectedType)) {
+            return PyCallable_as_functional_interface(env, pyobject, expectedType);
+        } else if (PyErr_Occurred()) {
+            return NULL;
+        } else if ((*env)->IsAssignableFrom(env, JPYCALLABLE_TYPE, expectedType)) {
+            return PyCallable_As_JPyCallable(env, pyobject);
+        }
+    } else if ((*env)->IsAssignableFrom(env, expectedType, JPYOBJECT_TYPE)) {
```

Comment:
This check is the opposite of the other – it's checking whether the *expected type* is `PyObject`, not whether we can *assign* a `PyObject` to the expected type (which is typically a fall-through case when expected type is `Object`).

Reply:
I think `IsSameObject` would be better since it doesn't work correctly on subclasses of PyObject.

Reply:
Good call – changed to `IsSameObject` (this allow means that the `PyCallable` check can be moved back down to its original position, since it will no longer trigger the `PyObject` branch)